src/common-components/ThirdPartyAuth.jsx

```diff
@@ -20,20 +24,31 @@ import {
 const ThirdPartyAuth = (props) => {
   const { formatMessage } = useIntl();
   const {
-    providers, secondaryProviders, currentProvider, handleInstitutionLogin, thirdPartyAuthApiStatus,
+    providers, 
+    secondaryProviders, 
+    currentProvider, 
+    handleInstitutionLogin, 
+    thirdPartyAuthApiStatus, 
+    isLoginPage,
   } = props;
   const isInstitutionAuthActive = !!secondaryProviders.length && !currentProvider;
   const isSocialAuthActive = !!providers.length && !currentProvider;
   const isEnterpriseLoginDisabled = getConfig().DISABLE_ENTERPRISE_LOGIN;
+  const enterpriseLoginURL = getConfig().LMS_BASE_URL + ENTERPRISE_LOGIN_URL;
 
   return (
     <>
       {((isEnterpriseLoginDisabled && isInstitutionAuthActive) || isSocialAuthActive) && (
         <div className="mt-4 mb-3 h4">
-          {formatMessage(messages['registration.other.options.heading'])}
+          {formatMessage(messages[isLoginPage ? 'login.other.options.heading' : 'registration.other.options.heading'])}
```

Comment:
i am not sure about it but is it fine to do conditioning while indexing array?
can we apply condition on format message ? 
Maybe
```
{isLoginPage 
? formatMessage(messages['login.other.options.heading']) 
: formatMessage(messages['registration.other.options.heading'])}
```

Reply:
I have applied your solution